Source/santad/EventProviders/EndpointSecurity/EnrichedTypes.h

```diff
@@ -0,0 +1,250 @@
+/// Copyright 2022 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+/// This file groups all of the enriched message types - that is the
+/// objects that are constructed to hold all enriched event data prior
+/// to being logged.
+
+#ifndef SANTA__SANTAD__EVENTPROVIDERS_ENDPOINTSECURITY_ENRICHEDTYPES_H
+#define SANTA__SANTAD__EVENTPROVIDERS_ENDPOINTSECURITY_ENRICHEDTYPES_H
+
+#include <time.h>
+#include <uuid/uuid.h>
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include "Source/santad/EventProviders/EndpointSecurity/Message.h"
+
+// Forward declarations
+namespace santa::santad::logs::endpoint_security::serializers {
+class Serializer;
+class BasicString;
+}  // namespace santa::santad::logs::endpoint_security::serializers
+
+namespace santa::santad::event_providers::endpoint_security {
+
+class EnrichedFile {
+ public:
+  EnrichedFile(std::optional<std::shared_ptr<std::string>> &&user,
+               std::optional<std::shared_ptr<std::string>> &&group,
+               std::optional<std::shared_ptr<std::string>> &&hash)
+      : user_(std::move(user)),
+        group_(std::move(group)),
+        hash_(std::move(hash)) {}
+
+  // Allow serializers to access internals
+  friend class santa::santad::logs::endpoint_security::serializers::BasicString;
```

Comment:
Instead of giving access to internals to the serializers (and the required use of friend, forward declarations _and_ adding these relationships in dozens of places when a new serializer is added) could we instead have getters for the fields they need access to, or just make the fields public?

Reply:
Refactored.